Source/Logger/trace.cpp

```diff
@@ -292,7 +302,17 @@ STDAPI_(void) HCTraceImplMessage_v(
     }
 
     // Only do work if there's reason to
-    if (GetTraceState().GetClientCallback() == nullptr && !GetTraceState().GetTraceToDebugger() && !GetTraceState().GetEtwEnabled())
+    bool haveClientCallback = false;
+    for (size_t i = 0; i < _countof(traceState.clientCallbacks); ++i)
+    {
+        if (traceState.clientCallbacks[i])
+        {
+            haveClientCallback = true;
+            break;
+        }
+    }
+
+    if (haveClientCallback && !traceState.GetTraceToDebugger() && !traceState.GetEtwEnabled())
```

Comment:
Shouldn't this check !haveClientCallback?

Reply:
It should, good catch.